src/Renci.SshNet/SshCommand.cs

```diff
@@ -294,6 +294,9 @@ public string EndExecute(IAsyncResult asyncResult)
                             _channel.Close();
                         }
 
+                        this._session.Disconnected -= Session_Disconnected;
+                        this._session.ErrorOccured -= Session_ErrorOccured;

```

Comment:
Something as simple as..

`namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            SshClient connection = new SshClient("127.0.0.1", "user", "password");
            connection.Connect();
            if (connection.IsConnected)
            {
                while (true)
                {
                    connection.RunCommand("pwd"); //leaks here
                }
            }
        }
    }
}`

will accumulate EventHandlers that are from "this._session" that keep SshCommands alive. Note that it's hard to see the memory usage from task manager from this program. My use case uses 10+ threads polling a lot and over the course of a few hours the memory usage climbs noticeably. If you use a program called ".Net Memory Profiler" it will tell you this information very neatly. It has a free trial for a few days, using this is how I discovered this particular issue.